### PR TITLE
Update Entity type references during merge

### PR DESCRIPTION
**Pull request by CNAChino**
# What Changed

# Why

Todo:

- [x] Add tests
- [ ] Add docs

### CODE REVIEW

File: src/main/java/com/intuit/graphql/orchestrator/schema/fold/UnifiedXtextGraphFolder.java [hunk 1]
```diff
@@ -103,7 +115,7 @@ private UnifiedXtextGraph merge(UnifiedXtextGraph accumulator, XtextGraph curren
           );
     }
 
-    resolveTypeConflicts(accumulator.getTypes(), current.getTypes(), current);
+    resolveTypeConflicts(accumulator.getTypes(), current.getTypes(), accumulator, current);
```
Comment (ashpak-shaikh) [inline src/main/java/com/intuit/graphql/orchestrator/schema/fold/UnifiedXtextGraphFolder.java:118]: can we just pass the accumulator and current to this method ?

File: src/main/java/com/intuit/graphql/orchestrator/schema/fold/UnifiedXtextGraphFolder.java [hunk 2]
```diff
@@ -148,10 +161,42 @@ private void resolveTypeConflicts(
...
+      mergeEntityTypes(existingType, incomingType, accumulator, incomingXtextGraph);
+    }
+  }
+
+  private void mergeEntityTypes(TypeDefinition existingType, TypeDefinition incomingType,
+      UnifiedXtextGraph accumulator, XtextGraph incomingXtextGraph) {
+    if(isBaseType(existingType) && isEntityExtensionType(incomingType)) {
+      // TODO consider calling EntityTypeMerging here
+      incomingXtextGraph.getTypes().put(existingType.getName(), existingType);
+      updateTypeReferencesInAnXtextGraph(incomingType, existingType, incomingXtextGraph);
```
Comment (ashpak-shaikh) [inline src/main/java/com/intuit/graphql/orchestrator/schema/fold/UnifiedXtextGraphFolder.java:182]: lets comment this line like we discussed and add an explanation for if...else block

File: src/main/java/com/intuit/graphql/orchestrator/schema/fold/UnifiedXtextGraphFolder.java (hunk 1 above)
Comment (CNAChino) [reply src/main/java/com/intuit/graphql/orchestrator/schema/fold/UnifiedXtextGraphFolder.java:117]: make sense will do

File: src/main/java/com/intuit/graphql/orchestrator/schema/fold/UnifiedXtextGraphFolder.java (hunk 2 above)
Comment (CNAChino) [reply src/main/java/com/intuit/graphql/orchestrator/schema/fold/UnifiedXtextGraphFolder.java]: done

File: src/test/groovy/com/intuit/graphql/orchestrator/integration/schemastitcher/EntityMultiServiceFieldReferences.groovy [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+                .collect(toList()) containsAll(["a1", "a2", "a3"])
+
+        GraphQLObjectType queryType = actualRuntimeGraph.getOperation(Operation.QUERY)
+        queryType.getFieldDefinition("topField1").getType() == typeA
+        queryType.getFieldDefinition("topField2").getType() == typeA
+        queryType.getFieldDefinition("topField3").getType() == typeA
+        queryType.getFieldDefinition("topField4").getType() == typeA
+
+        where:
+        SCHEMA_A                     | SCHEMA_B
```
Comment (CNAChino) [inline src/test/groovy/com/intuit/graphql/orchestrator/integration/schemastitcher/EntityMultiServiceFieldReferences.groovy:69]: This is where have the base type or extension type goes into the accumulator first

File: src/main/java/com/intuit/graphql/orchestrator/schema/fold/UnifiedXtextGraphFolder.java (hunk 1 above)
Comment (CNAChino) [reply src/main/java/com/intuit/graphql/orchestrator/schema/fold/UnifiedXtextGraphFolder.java]: done

**Review by ashpak-shaikh — APPROVED** [review]

**Review by ashpak-shaikh — APPROVED** [review]

**Pull request merged**